src/main/java/tds/testpackageconverter/converter/mappers/CsidEnhancer.java

```diff
@@ -0,0 +1,60 @@
+package tds.testpackageconverter.converter.mappers;
+
+import org.opentestsystem.shared.contentspecid.ContentSpecId;
+import org.opentestsystem.shared.contentspecid.ContentSpecIdConverter;
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecFormat;
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecGrade;
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecSubject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+/**
+ * Enhance legacy format IDs to enhanced format content spec IDs. Since the legacy IDs in the test
+ * packages don't contain specification version information, it is assumed that when the subject is Math,
+ * the IDs are in MATH, v6 format, and if ELA, then the IDs are in ELA, v1. (For ELA, v1 is the only current
+ * legacy format.)
+ *
+ * The grade level also must be provided because some legacy IDs omit this information.
+ */
+public class CsidEnhancer {
+    private static final Logger log = LoggerFactory.getLogger(LegacyScoringTestPackageFormMapper.class);
+
+    private ContentSpecIdConverter converter = new ContentSpecIdConverter();
+
+    private final ContentSpecSubject subject;
+    private String prefix;
+    private final ContentSpecGrade grade;
+
+    public CsidEnhancer(String subject, String grade) {
+        if (subject.equalsIgnoreCase("MATH")) {
+            this.subject = ContentSpecSubject.MATH;
+            this.prefix = ContentSpecFormat.MATH_V6.getLegacyPrefix();
+        } else if (subject.equalsIgnoreCase("ELA")) {
+            this.subject = ContentSpecSubject.ELA;
+            this.prefix = ContentSpecFormat.ELA_V1.getLegacyPrefix();
+        } else {
```

Comment:
I feel this is an Exceptional and invalid input.  The `testPackageConverter` should work with only Math or ELA test packages otherwise it should fail.  I recommend throwing an IllegalArgumentException (or appropriate Exception) instead.

Reply:
Hmm, how strongly do you feel about this one? I agree that having a subject not Math or ELA should probably be a hard error, but I don't like having this being the place to discover that and blow up the conversion process.